### ISSUE TITLE
[JENKINS-12028] A path argument for launching jenkins.war file causes an exception and recursively deletes all files from the current folder

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#16301]
<p>Starting Jenkins via the war file and supplying a path as argument for the application, causes an exception and recursively deletes all files from the current folder.</p>

<p>Steps:<br/>
1. Download the jenkins.war file from <a href="https://jenkins-ci.org/" class="external-link" target="_blank" rel="nofollow noopener">https://jenkins-ci.org/</a><br/>
2. Start jenkins via: 'java -jar jenkins.war ./'<br/>
3. Stop the application with Ctrl+C</p>

<p>After step 3 check the current folder and notice that all files and sub folders have been deleted.</p>

<p>The exception I get is:</p>

<p>$ java -jar jenkins.war ./<br/>
Running from: /Volumes/data/jenkins/jenkins.war<br/>
webroot: $user.home/.jenkins<br/>
<span class="error">&#91;Winstone 2011/12/07 17:09:05&#93;</span> - Beginning extraction from war file<br/>
<span class="error">&#91;Winstone 2011/12/07 17:09:05&#93;</span> - Error initializing web application: prefix []<br/>
java.util.zip.ZipException: error in opening zip file<br/>
	at java.util.zip.ZipFile.open(Native Method)<br/>
	at java.util.zip.ZipFile.<init>(ZipFile.java:127)<br/>
	at java.util.jar.JarFile.<init>(JarFile.java:135)<br/>
	at java.util.jar.JarFile.<init>(JarFile.java:99)<br/>
	at winstone.HostConfiguration.getWebRoot(HostConfiguration.java:253)<br/>
	at winstone.HostConfiguration.<init>(HostConfiguration.java:73)<br/>
	at winstone.HostGroup.initHost(HostGroup.java:85)<br/>
	at winstone.HostGroup.<init>(HostGroup.java:45)<br/>
	at winstone.Launcher.<init>(Launcher.java:197)<br/>
	at winstone.Launcher.main(Launcher.java:398)<br/>
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/>
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)<br/>
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)<br/>
	at java.lang.reflect.Method.invoke(Method.java:597)<br/>
	at Main._main(Main.java:268)<br/>
	at Main.main(Main.java:96)</p>

<p><span class="error">&#91;Winstone 2011/12/07 17:09:05&#93;</span> - HTTP Listener started: port=8080<br/>
<span class="error">&#91;Winstone 2011/12/07 17:09:05&#93;</span> - AJP13 Listener started: port=8009<br/>
<span class="error">&#91;Winstone 2011/12/07 17:09:05&#93;</span> - Winstone Servlet Engine v0.9.10 running: controlPort=disabled</p>

<p>Whatever argument is specified it has to be handled gracefully and should not result in deleting whole folders on disk. I think that's a critical issue and that I was lucky to run the war file from a newly created 'jenkins' sub folder. Otherwise I would have wiped out my whole data partition.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12028" target="_blank">A path argument for launching jenkins.war file causes an exception and recursively deletes all files from the current folder</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core
<li><b>resolution</b>: Incomplete
<li><b>resolved</b>: 2014-06-14T21:16:03+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Starting Jenkins via the war file and supplying a path as argument for the application, causes an exception and recursively deletes all files from the current folder.</p>

<p>Steps:
1. Download the jenkins.war file from <a href="https://jenkins-ci.org/" class="external-link" target="_blank" rel="nofollow noopener">https://jenkins-ci.org/</a>
2. Start jenkins via: 'java -jar jenkins.war ./'
3. Stop the application with Ctrl+C</p>

<p>After step 3 check the current folder and notice that all files and sub folders have been deleted.</p>

<p>The exception I get is:</p>

<p>$ java -jar jenkins.war ./
Running from: /Volumes/data/jenkins/jenkins.war
webroot: $user.home/.jenkins
<span class="error">&#91;Winstone 2011/12/07 17:09:05&#93;</span> - Beginning extraction from war file
<span class="error">&#91;Winstone 2011/12/07 17:09:05&#93;</span> - Error initializing web application: prefix []
java.util.zip.ZipException: error in opening zip file
	at java.util.zip.ZipFile.open(Native Method)
	at java.util.zip.ZipFile.&lt;init&gt;(ZipFile.java:127)
	at java.util.jar.JarFile.&lt;init&gt;(JarFile.java:135)
	at java.util.jar.JarFile.&lt;init&gt;(JarFile.java:99)
	at winstone.HostConfiguration.getWebRoot(HostConfiguration.java:253)
	at winstone.HostConfiguration.&lt;init&gt;(HostConfiguration.java:73)
	at winstone.HostGroup.initHost(HostGroup.java:85)
	at winstone.HostGroup.&lt;init&gt;(HostGroup.java:45)
	at winstone.Launcher.&lt;init&gt;(Launcher.java:197)
	at winstone.Launcher.main(Launcher.java:398)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at Main._main(Main.java:268)
	at Main.main(Main.java:96)</p>

<p><span class="error">&#91;Winstone 2011/12/07 17:09:05&#93;</span> - HTTP Listener started: port=8080
<span class="error">&#91;Winstone 2011/12/07 17:09:05&#93;</span> - AJP13 Listener started: port=8009
<span class="error">&#91;Winstone 2011/12/07 17:09:05&#93;</span> - Winstone Servlet Engine v0.9.10 running: controlPort=disabled</p>

<p>Whatever argument is specified it has to be handled gracefully and should not result in deleting whole folders on disk. I think that's a critical issue and that I was lucky to run the war file from a newly created 'jenkins' sub folder. Otherwise I would have wiped out my whole data partition.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Issue has been found on OS X 10.7.2 by using the jenkins.war file downloaded from <a href="https://jenkins-ci.org/">https://jenkins-ci.org/</a></code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12028?focusedId=203919&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-203919">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Needs to be reproduced with Jetty.</p>

<p>Just file a new issue if this still occurs on current Jenkins releases.</p>
</pre>
</details>

</ul>
</details>

<p>Needs to be reproduced with Jetty.</p>

<p>Just file a new issue if this still occurs on current Jenkins releases.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/whimboo.jpg" title="whimboo's avatar" /> whimboo:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-12028?focusedId=204047&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-204047">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This seems to work fine with Jetty. I cannot reproduce it anymore with version 1.554.2.</p>
</pre>
</details>

</ul>
</details>

<p>This seems to work fine with Jetty. I cannot reproduce it anymore with version 1.554.2.</p>
